provider/cmd/pulumi-resource-nuage/nuage_provider/serverless_database.py

```diff
@@ -99,10 +111,10 @@ def __init__(
         )
 
         cluster = aws.rds.Cluster(
-            resource_name=f"{args.resource_name}-cluster",
-            cluster_identifier_prefix=f"{name}-cluster-",
-            database_name=args.database_name or name,
-            master_username=args.master_username or name,
+            resource_name=resource_name,
+            cluster_identifier_prefix=self.name_.apply(lambda name: f"{name}-cluster-"),
+            database_name=args.database_name or self.name_,
+            master_username=args.master_username or self.name_,
```

Comment:
I think I'd just go for args.username 

Creating a username implicitly is kinda dangerous security wise